### PR TITLE
Revert to using quiescing helper

### PR DESCRIPTION
**Pull request by adam-fowler**
This means we can quiesce channels outside of the handle task. Required for channels which involve an upgrade path before passing to concurrency handlers.

### CODE REVIEW

**Comment by adam-fowler** [comment]
This is also a minor optimisation

File: Sources/HummingbirdCore/Server/Server.swift [hunk 1]
```diff
@@ -272,6 +291,17 @@ public actor Server<ChildChannel: ServerChildChannel>: Service {
...
+    /// Initialize the `ServerSocketChannel` with `initializer`. The most common task in initializer is to add
+    /// `ChannelHandler`s to the `ChannelPipeline`.
+    ///
+    /// The `ServerSocketChannel` uses the accepted `Channel`s as inbound messages.
+    ///
+    /// - note: To set the initializer for the accepted `SocketChannel`s, look at `ServerBootstrap.childChannelInitializer`.
+    ///
+    /// - parameters:
+    ///     - initializer: A closure that initializes the provided `Channel`.
+    func serverChannelInitializer(_ initializer: @escaping @Sendable (Channel) -> EventLoopFuture<Void>) -> Self
```
Comment (Joannis) [inline Sources/HummingbirdCore/Server/Server.swift:303]: Am I missing an implementation for HTTP/1.1 and HTTP/2?

Comment (adam-fowler) [reply Sources/HummingbirdCore/Server/Server.swift:303]: Not sure what you mean here? This is here to abstract what bootstrap I am using eg Posix or TS

**Comment by adam-fowler** [comment]
This fixes #503

**Comment by Joannis** [comment]
Can we get a test to reproduce #503 in this PR?

**Comment by adam-fowler** [comment]
Was just thinking that

**Comment by adam-fowler** [comment]
> Can we get a test to reproduce #503 in this PR?

Added `HummingbirdTLSTests.testGracefulShutdownWithDanglingConnection`

**Review by Joannis — APPROVED** [review]

**Pull request merged**